packages/bpk-component-infinite-scroll/src/withInfiniteScroll.js

```diff
@@ -138,32 +145,56 @@ const withInfiniteScroll = (
           index: 0,
           elementsPerScroll: this.props.elementsPerScroll,
           elementsToRender: [],
-        }).then(newState => {
-          this.setState({
-            ...newState,
+        })
+          .then(newState => {
+            this.setState({
+              ...newState,
+            });
+          })
+          .catch(() => {
+            this.handleFetchItemsFailed();
           });
-        });
       }
     }
 
     componentWillUnmount() {
       this.props.dataSource.removeListener(this.updateData);
+      this.unobserveSentinel();
+    }
+
+    unobserveSentinel() {
       if (this.sentinel) {
         this.observer.unobserve(this.sentinel);
       }
     }
 
+    handleFetchItemsFailed() {
+      this.unobserveSentinel();
+      this.setState({ isListFinished: true });
+    }
+
     updateData = () => {
       const { index } = this.state;
-      this.props.dataSource.fetchItems(0, index).then(updatedElements => {
-        this.setState({
-          elementsToRender: updatedElements,
+      const { dataSource } = this.props;
+      dataSource
+        .fetchItems(0, index)
+        .then(updatedElements => {
+          this.setState({
+            elementsToRender: updatedElements,
+          });
+        })
+        .catch(() => {
+          this.handleFetchItemsFailed();
```

Comment:
```suggestion
        .catch(this.handleFetchItemsFailed);
```
Bit shorter and has the same effect.